org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/command/AnalyzeCommand.java

```diff
@@ -75,9 +78,20 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 
   private static void runAnalysisJob(Shell shell, Map<ISonarLintProject, Collection<FileWithDocument>> filesPerProject) {
     var totalFileCount = filesPerProject.values().stream().mapToInt(Collection::size).sum();
-    if (totalFileCount > 1 && !askConfirmation(shell)) {
+    
+    if (!SonarLintGlobalConfiguration.ignoreEnhancedFeatureNotifications()) {
+      MessageDialogUtils.enhancedWithConnectedModeInformation(shell, "Are you working with a CI/CD pipeline?",
+        "Running an analysis with SonarQube / SonarCloud in your pipeline might be better suited for analyzing "
+        + "multiple files or a whole project!");
+    } else if (totalFileCount > 10 && !askConfirmation(shell)) {
```

Comment:
This is the reason for the changes in the `java-simple` IT. Marco was also saying to change the constraint because of the noise.

Reply:
This change could have been made in a separate ticket/commit, but fair enough.